ratis-server-api/src/main/java/org/apache/ratis/server/RaftServer.java

```diff
@@ -249,5 +262,11 @@ public Builder setParameters(Parameters parameters) {
       this.parameters = parameters;
       return this;
     }
+
+    // TODO(jiacheng): add UTs
+    public Builder setUncaughtExceptionHandler(Thread.UncaughtExceptionHandler exceptionHandler) {
```

Comment:
Adding an optional field to `RaftServer.Builder` already gives backward compatibility. Existing usages don't need to change anything if they upgrade Ratis, it's just they will be using the default `UncaughtExceptionHandler`